pkg/internal/validation/dns.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// This file's code was modified from "k8s.io/apimachinery/pkg/util/validation"
+// to avoid package dependencies. In case of additional functionality from
+// "k8s.io/apimachinery" is needed, re-consider whether to add the dependency.
+
+const (
+	dns1123LabelFmt     string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+	dns1123SubdomainFmt string = dns1123LabelFmt + "(\\." + dns1123LabelFmt + ")*"
+	dns1035LabelFmt     string = "[a-z]([-a-z0-9]*[a-z0-9])?"
+)
+
+type dnsValidationConfig struct {
+	format   string
+	maxLen   int
+	re       *regexp.Regexp
+	errMsg   string
+	examples []string
+}
+
+var dns1123LabelConfig = dnsValidationConfig{
+	format:   dns1123LabelFmt,
+	maxLen:   56, // = 63 - len("-system")
```

Comment:
I think the `validation` package should not know about the `"-system"` prefix, the method should be called with that already prefixed and the `maxLen` field should be left to 63.

Reply:
Just to clarifies: The project name will be used as part of the namespace added in the files.
So, it will be "<projectname-system>". In this way, when we scaffold the project we cannot allow a projectName with more than  56 characters. However, I agree that how it will be implemented as a hall for improvements like you described. 

However, it may not part of the context of this PR as well.